android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/Worker.kt

```diff
@@ -22,6 +22,16 @@ package com.uber.rib.core
  * other lifecycles we're interested in.
  */
 public interface Worker {
+
+  /**
+   * Specifies in which ThreadingType the WorkerBinder will bound this Worker.
+   *
+   * Whe not overriden, will use the thread where [WorkerBinder.bind] call happens
+   */
+  @JvmDefault
+  public val workerBinderThreadingType: WorkerBinderThreadingType
+    get() = WorkerBinderThreadingType.CALLER_THREAD
```

Comment:
Assuming we wanted to default to background to prevent any new additions from being made unless the explicitly indicated calling thread, what would that look like and how could we migrate internally (ie, new internal base class with a linter against it and baseline it)? Anything reasonable that doesn't result in a massive migration?

Reply:
Was thinking we could rely on the new listener to trigger non-fatals for expensive UI workers (e.g. crash on debug builds) and have global monitoring that will report expensive UI workers. 

Given that @psteiger is proposing a new RibCoroutineWorker (in https://github.com/uber/RIBs/pull/547)  that will run by default on the background, internally we could recommend to use this new definition (with suspend function), mark as deprecated this existing old definition and keep it for backward compatibility with existing usages